### DISCUSSION TITLE
Is it possible to make something that has more Unicode than the 16 bit set?

### DISCUSSION THREAD

**Discussion by snowboyz0825** [TurboWarp/extensions#1347] [General]
Whilst experimenting, I figured out that on any extension with Unicode, it starts to repeat itself after 65536 characters. Using Google, I found there are ~149000 characters. After a little more researching, I found that it was the 16 bit integer limit. I believe in the 16 bit set there are only 65536 characters, and I was wondering if there was an extension that would let me pull from a more modern version of it.

**Comment by CST1229** [comment]
UTF-16 can support all Unicode characters, though characters above 65k will have to be represented with multiple 16-bit "codepoints" joined together (which I assume most extensions don't account for, including mine).

**Reply by snowboyz0825** [reply]
It did not, and I used text as well as encoding to try it. After 65k it will begin to repeat, and specifically for what my project was doing it won't work

**Reply by GarboMuffin** [reply]
Maybe you should share your actual code

**Reply by CST1229** [reply]
I mean that, in the UTF-16 character encoding, for example the 😎 emoji is represented as two 16-bit numbers: 55357 and 56846, next to eachother. So you might have to do something like this:
![](https://github.com/TurboWarp/extensions/assets/68464103/0302b189-5ce8-4407-bd00-edf87996644e)

(I definitely do think that having it actually support the full correct Unicode character number range instead of just supporting 0-65536 with UTF-16 encoding makes more sense, but that'll probably be in new blocks to not break compatibility.)

**Reply by snowboyz0825** [reply]
![block_3_8_2024-11_39_51 AM](https://github.com/TurboWarp/extensions/assets/101284660/b6339fb4-3807-451c-a5e2-2cbc24b93deb)
Adds the Unicode to the list and then checks for any repeats. Originally I wanted a 10000-character long list as that's the limit for scratch (as far as I'm aware). It worked and I essentially used it to compress any 4 numbers into a single Unicode character, which is useful for copy/paste saving. Now I want to see if I can push it to the limit and possibly have it compress 5 numbers at once. I needed to check because scratch counts some strange unicode characters as capitals of the others, and this makes sure nothing is. I would need a list of 100000 Unicode characters, which I should be able to squeeze into 10000 10-character strings. After observing it for a while, it started always recognizing repeats at exactly 65536 characters away, and if you put in for example 65 and 65601 (65536+65) they are both capital A. Admittedly I don't know how Unicode works, my research was only surface deep. I hope this helps, and I'll try what CST mentioned above

**Reply by snowboyz0825** [reply]
Oh also if anyone ever is wondering for some reason, I did find a 10000 long list at 11506-21506
